common/static/js/lib/pretty-print.js

```diff
@@ -0,0 +1,97 @@
+/**
```

Comment:
Also, if we're taking over the code, I think you should add unit tests to verify that it behaves as expected. The code is very hard to read, IMO, which makes me nervous that there are lurking bugs.